configure.ac

```diff
@@ -1229,33 +1229,19 @@ SQUID_CHECK_LIB_WORKS(gss,[
 ])
 
 SQUID_AUTO_LIB(ldap,[LDAP],[LIBLDAP])
+dnl On MinGW set Windows LDAP libraries using -lwldap32
+AS_IF([test "x$with_ldap" != "xno" -a "$squid_host_os" = "mingw"],[
+  LIBLDAP_LIBS="$LIBLDAP_LIBS -lwldap32"
```

Comment:
Maybe, for now I am leaving the logic inside this case as-was. Testing changes to it need some testing on MinGW which we are not quite up to doing (yet)

Reply:
That excuse itself is very weak, but when combined with the fact that this PR does change the condition, it becomes invalid.

AFAICT, the PRs in this series routinely introduce untested changes. That is just the nature of this work, given our current CI limitations. IMO, we should not apply "need some testing before I modify this further" logic when dealing with changes that appear to violate basic rules. Given the two evils, I recommend increasing the risk of breaking MinGW build instead of committing such violations. These changes should be moving us forward as far as code quality is concerned. We can fix marginal builds later, as needed.